File: src/garanti/internal/executor_ffi.erl
-module(executor_ffi).
-export([run_catching/1]).

% Calls the given test function inside a try/catch so that any exception (e.g. a
% Gleam panic) is caught before it propagates. When an exception is caught the
% process exits normally rather than crashing, which suppresses the Erlang crash
% report that would otherwise be printed to stdout.
%
% Returns {ok, Result} if the function completes, or {error, nil} if it throws.
run_catching(TestFn) ->
    try
        {ok, TestFn()}
    catch
        _:_ -> {error, nil}
    end.

File: garanti_javascript/src/garanti_javascript/internal/executor_ffi.mjs
import { Ok, Error } from "../../gleam.mjs";

// Run the test function without raising an exception (to not abort a test run). This
// is similar to the Erlang executor (garanti_erlang/src/internal/executor_ffi.erl).
export function run_catching(testFn) {
  try {
    return new Ok(testFn());
  }
  catch (_error) {
    return new Error(undefined);
  }
}
